xova/apps/xova/fixvis.py

```diff
@@ -71,45 +77,39 @@ def dense2sparce_uvw(a1, a2, time, ddid, padded_uvw):
         ddid: sparse data discriptor index
         padded_uvw: a dense ddid-less uvw matrix
                     returned by synthesize_uvw of shape
-                    (ntime * nbl, 3), fastest varying 
+                    (ntime * nbl, 3), fastest varying
                     by baseline, including auto correlations
     """
     assert time.size == a1.size
     assert a1.size == a2.size
     ants = np.concatenate((a1, a2))
-    unique_ants = np.unique(ants)
+    unique_ants = np.arange(np.max(ants) + 1)
```

Comment:
this caused problems when chunks were so small not all antennae were in the chunk. This fix works around this old indexing problem

Reply:
I'm not sure that this will handle all cases since it depends on the max antenna being passed in a1 and a2. If this doesn't work `np.unique` I don't think it will work for `np.arange(np.max(ants) + 1)` either?

Reply:
This is to do with the formula (I've checked and tested why it broke on my lunatic polarimetry uvw regeneration code, where I caught this originally. The differences of differences method I used to derive the formula for baseline unique index assumes that na is greater or equal to the maximum antenna index loaded in through a1 and a2. Using the size of unique ants breaks this assumption and you get negative indexes out of the formula. Therefore the max + 1 criterion. If you have a better way of computing baseline unique indexes let me know.

Reply:
Ah OK, I understand now